presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -191,16 +228,22 @@ public GroupedAccumulator createGroupedIntermediateAccumulator()
         }
     }
 
-    @Override
-    public boolean hasOrderBy()
+    private GroupedAccumulator createSpillableGroupedAccumulator()
     {
-        return !orderByChannels.isEmpty();
+        GroupedAccumulator accumulator = createDefaultGroupedAccumulator();
+        if (!hasDistinct() && !hasOrderBy()) {
+            return accumulator;
+        }
+
+        return new SpillableFinalOnlyGroupedAccumulator(sourceTypes, (FinalOnlyGroupedAccumulator) accumulator);
     }
 
-    @Override
-    public boolean hasDistinct()
+    private GroupedAccumulator createSpillableGroupedIntermediateAccumulator()
     {
-        return distinct;
+        if (!hasOrderBy() && !hasDistinct()) {
+            return createDefaultGroupedIntermediateAccumulator();
+        }
+        return createSpillableGroupedAccumulator();
```

Comment:
Is this right? the function is to create IntermediateAccumulator but here it delegates to GroupedAccumulator and then use createDefaultGroupedAccumulator instead of createDefaultGroupedIntermediateAccumulator?

Reply:
Yes since we do the same logic in either case since we do not care about the underlying accumulator working on intermediate values in our FinalOnlyGroupedAccumulator/SpillableFinalOnlyGroupedAccumulator delegate. If I was to separate these functions the resulting logic should be the same. It originally was separate but I noticed it could be simplified.